src/client/util.ts

```diff
@@ -10,9 +10,15 @@ export class Util {
         const globalCommands = registeredCommands.filter(x => !x.ServerId);
         const guildCommands = registeredCommands.filter(x => x.ServerId);
 
-        const globalCommandData: SlashCommandBuilder[] = globalCommands
-            .filter(x => x.Command.CommandBuilder)
-            .flatMap(x => x.Command.CommandBuilder);
+        const globalCommandData: SlashCommandBuilder[] = [];
+
+        for (let command of globalCommands) {
+            if (!command.Command.CommandBuilder) continue;
+
+            if ((command.Environment & CoreClient.Environment) == CoreClient.Environment) {
```

Comment:
Should use `environment &= CoreClient.Environment` to be consistent with code in `client.ts`